llvm/lib/Target/SPIRV/SPIRVUtils.cpp

```diff
@@ -251,6 +251,24 @@ SPIRV::MemorySemantics::MemorySemantics getMemSemantics(AtomicOrdering Ord) {
   llvm_unreachable(nullptr);
 }
 
+SPIRV::Scope::Scope getMemScope(const LLVMContext &Ctx, SyncScope::ID ID) {
+  SmallVector<StringRef> SSNs;
+  Ctx.getSyncScopeNames(SSNs);
+
+  StringRef MemScope = SSNs[ID];
+  if (MemScope.empty() || MemScope == "all_svm_devices")
```

Comment:
Just avoid all_svm_devices altogether? It's the same as the default / ""

Reply:
I personally prefer to keep an explicit check for whatever default string we have. It should be easier (and hence more error-prone) to reuse names from https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#_scope_id instead of thinking about, what is LLVM's default and how it maps on SPIR-V scopes.

Reply:
On consideration, I think I'm with @arsenm on this one, both `singlethread` and the empty string (system) have a defined meaning in LLVM, with all else being target specific. So, IMHO, for these we actually should think about LLVM's default, and how the target implements those, whilst for everything else we should defer to the target.